Reorder param to avoid breaking change, add first example for ADL

begoldsm opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

Commit:
fix for validation error in ADLA

Remove minimum and add an explanation.

Commit:
Initial example for feedback

Please look at this and let me know if this is the right approach

Commit:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

Commit:
move firewallallowazureIps to the end

This avoids a param reordering breaking change.

olydis commented:
There are model validation issues: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/200526483

> In operation "Job_Create", parameter jobIdentity is required in the swagger spec but is not present in the provided example parameter values.

> Missing required property: script

> Additional properties not allowed: expirationTimeUtc,totalRunningTime,totalQueuedTime,totalPausedTime,totalCompilationTime,errorSource,compileMode,algebraFilePath,rootProcessNodeId,resources,owner

Commit:
fix example.

olydis commented:
Operations API must be implemented for the service (applies to both swagger files). Example: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-cdn/2016-10-02/swagger/cdn.json#L1462

olydis commented:
Usage of Guid is not recommended in body parameters. If GUIDs are absolutely required in your service, please get sign off from the Azure API review board.

Commit:
Unlink example

This is due to github issue:
https://github.com/Azure/azure-rest-api-specs/issues/928

olydis reviewed (approved):
All issues addressed (see https://github.com/Azure/azure-rest-api-specs/pull/897 for plans about uuid issue; Operations API must become milestone work item), good to merge!

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby.svg?branch=RestAPI-PR927)](https://travis-ci.org/AutorestCI/azure-sdk-for-ruby) https://github.com/Azure/azure-sdk-for-ruby/pull/686

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-node.svg?branch=RestAPI-PR927)](https://travis-ci.org/AutorestCI/azure-sdk-for-node) https://github.com/Azure/azure-sdk-for-node/pull/2059

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR927)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/991
